front50-web/src/test/groovy/com/netflix/spinnaker/front50/controllers/StrategyControllerTck.groovy

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package com.netflix.spinnaker.front50.controllers
+
+
+import com.amazonaws.ClientConfiguration
+import com.amazonaws.services.s3.AmazonS3Client
+import com.netflix.spinnaker.front50.model.S3PipelineStrategyDAO
+import com.netflix.spinnaker.front50.model.pipeline.Pipeline
+import com.netflix.spinnaker.front50.model.pipeline.PipelineStrategyDAO
+import com.netflix.spinnaker.front50.pipeline.StrategyRepository
+import com.netflix.spinnaker.front50.utils.CassandraTestHelper
+import com.netflix.spinnaker.front50.utils.S3TestHelper
+import rx.schedulers.Schedulers
+import spock.lang.IgnoreIf
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Subject
+import spock.lang.Unroll
+
+import java.util.concurrent.Executors
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.springframework.http.MediaType
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.setup.MockMvcBuilders
+
+abstract class StrategyControllerTck extends Specification {
```

Comment:
Sad face that we have a StrategyController and PipelineController and Pipeline/Strategy objects that are essentially identical. 

This is historical and not something I'll be able to change in this PR.